### PR TITLE
refactor: add media api for serving protected user files

### PR DESCRIPTION
**Pull request by macintoshpie**
#### What's this PR do?
Adds permissioning in front of serving media files (ie user uploaded files).
More specifically, creates `/api/v3/media/<filepath>` (and sets it as the MEDIA_URL) which verifies the requesting user has access to the file, before instructing Nginx to serve the file using the X-Accel-Redirect header. If the user doesn't have access or the file doesn't exist, it 404s.

#### How should this be manually tested?
If using docker-compose setup, you must be using the non-dev version (ie docker-compose.local). This is because this update requires Nginx to serve the files.
Create two separate orgs with different user accounts
Try to access different media files while logged in as different users (or not logged in). Media file locations:
- /api/v3/media/uploads
    - (used for most property imports)
    - create by importing from a csv
- /api/v3/media/buildingsync_files/
    - (used for bsync imports)
    - create by importing bsync file
- /api/v3/media/analysis_input_files/<analysis id>/
    - (where input files for an analysis are stored)
    - create by running bsyncr analysis. (for refresher, [ see this PR](https://github.com/SEED-platform/seed/pull/2547))
- /api/v3/media/analysis_output_files/
    - (where analysis output files are stored)
    - create by running bsyncr analysis

Try to access `<host>/media/<some file>` directly and it should fail (/media is no longer the MEDIA_URL). Try to access `<host>/protected/<some file>` and it should fail (the protected location is _internal_ to Nginx, meaning it's only accessible when not requested externally, ie in client request)

Try to break it by using it in unexpected ways...

#### What are the relevant tickets?
#2626

### CODE REVIEW

File: seed/views/v3/uploads.py [hunk 1]
```diff
@@ -29,6 +29,14 @@
 _log = logging.getLogger(__name__)
 
 
+def get_upload_path(filename):
```
Comment (macintoshpie) [inline seed/views/v3/uploads.py:32]: moved this out to use with testing

**Comment by axelstudios** [comment]
Nice work - looks like file uploads to `/seed/media/uploads/*` don't get listed under files?  We probably want to add a download icon to each column in the history that specifies `ImportFile` (and probably also next to each upload on the datasets page)

**Comment by macintoshpie** [comment]
TODO: when checking ImportFiles, make sure `deleted=False`

**Comment by macintoshpie** [comment]
TODO: add download links for files (e.g. property detail, data set, etc)

**Comment by macintoshpie** [comment]
@adrian-lara Should be ready for your review now. Let me know if you need help getting bsyncr analyses up and running again, I know it can be a pain

**Review by adrian-lara — APPROVED** [review]
Code changes look good, and happy + sad paths seem to be working as expected! 👍🏽

**Pull request merged**